### PR TITLE
adds examples for with_dir() and rebuilds documentation

### LINKED ISSUE #97: Add examples for `with_dir()`
Maybe showing switching to the temp directory and back or similar?

Wild caught usage examples in the [GitHub search results for 'with_dir'](https://github.com/search?q=with_dir+user%3Acran&type=Code).

### PR DESCRIPTION
**Pull request by dragosmg**
closes #97

### CODE REVIEW

**Comment by hadley** [comment]
I've updated the master branch — once https://travis-ci.org/r-lib/withr/builds/555672206 build successfully, can you please update this PR from master? The easiest way to do that is `usethis::pr_pull_upstream()`, then `usethis::pr_push()`. That should fix the build failure and eliminate the extraneous doc changes.

File: R/dir.R [hunk 1]
```diff
@@ -12,6 +12,14 @@ NULL
 #' @inheritParams with_collate
 #' @seealso [setwd()]
 #' @export
+#' @examples
+#' getwd() # current working directory
+#' 
+#' # change the current working directory to a temporary one and run getwd()
+#' with_dir(tempdir(), getwd())
+#' 
+#' # once the function execution is completed the working directory is restored
```
Comment (hadley) [inline R/dir.R]: I think you can eliminate this example

File: R/dir.R [hunk 2]
```diff
@@ -12,6 +12,14 @@ NULL
 #' @inheritParams with_collate
 #' @seealso [setwd()]
 #' @export
+#' @examples
+#' getwd() # current working directory
```
Comment (hadley) [inline R/dir.R]: I think the comments are slightly too verbose — I think you can assume that the reader already knows what `getwd()` does.

File: R/dir.R (hunk 1 above)
Comment (dragosmg) [reply R/dir.R]: @hadley You mean the whole thing? I think the travis build is still failing.

Comment (hadley) [reply R/dir.R]: Sorry, I meant just the last line, showing that the directory change was only temporary

File: R/dir.R [hunk 3]
```diff
@@ -12,6 +12,10 @@ NULL
 #' @inheritParams with_collate
 #' @seealso [setwd()]
 #' @export
+#' @examples
```
Comment (hadley) [inline R/dir.R]: Perfect!

Comment (dragosmg) [reply R/dir.R]: Thanks!

**Comment by hadley** [comment]
Thanks @dragosmg!

**Pull request merged**